openapi-codegen/core/src/main/scala/sttp/tapir/codegen/BasicGenerator.scala

```diff
@@ -61,6 +61,29 @@ object BasicGenerator {
            |}""".stripMargin
         headTag -> taggedObj
     }
+
+    val maybeSpecificationExtensionKeys = doc.paths
+      .flatMap { p =>
+        p.specificationExtensions.toSeq ++ p.methods.flatMap(_.specificationExtensions.toSeq)
+      }
+      .groupBy(_._1)
+      .map { case (keyName, pairs) =>
+        val values = pairs.map(_._2)
+        val distinctTypes = values.map(_.tpe).distinct
+        if (distinctTypes.size != 1)
```

Comment:
Actually, this check would break gratuitously if the value type was sometimes an empty map or empty list... Or if the values were sometimes homogeneous and sometimes heterogeneous. Will need a bit more sophistication...